### ISSUE TITLE
Null reference exception in Esent.cs when loading big IFC files

### ISSUE THREAD

**Issue by PIUM-arc** [xBimTeam/XbimEssentials#383]
Hello,

I've downloaded _XbimWindowsUI_ to do some test with _Xbim.WinformsSample_. I run the solution in a virtual machine and when I try to open IFC files bigger than 100 MB I have a Null reference exception on InstanceCache in _EsentModel.cs_.

The strange thing is, on another VM with the same OS and with the same XbimWindowsUI code, I don't reproduce the exception and don't have any issue to open the same IFC file. 

I have the same NuGet packages on both VM ( from nuget.org) but don't have the same behavior. 

Do you have any idea what could be the reason for this kind of issue ? 

Thank you.

**Comment by martin1cerny** [comment]
100 MB is a relatively small file. Xbim Toolkit can normally handle much bigger file. In fact, it should only be limited by the hardware capabilities. I'm happy to investigate the issue, but you would have to provide a repeatable sample code demonstrating the issue.

**Comment by PIUM-arc** [comment]
I'm using the XbimWindowsUI-master package, I didn't change the code of this solution, I just updated the NuGet packages with nuget.org versions and that's all.
The strange thing is if I copy paste my code from the VM where it works to the other VM, I will have the null reference exception, so with the exact same code I don't have the same behavior. 

I also tried to copy paste the Xbim.WinformsSample bin folder from the good VM to the bad one and execute the Xbim.WinformsSample.exe, and when I try to open a file bigger than 100 MB, I have a exception " Could not load file or assembly 'Esent.Interop, version= 1.9.4.0.  etc"

I'm kind of lost here because I have the same packages versions, at the same place on my two VMs , so why in one case I have an assembly problem (or null reference exception) ?

**Comment by martin1cerny** [comment]
Can you provide us with a simple code and data example to reproduce the issue?

**Comment by PIUM-arc** [comment]
I don't know what I could provide you, it's basically Xbim.WindowsUI with updated NuGet packages,I didn't change anything in the code, and I run Xbim.WinformsSample to open the IFC file. 

The strange thing is, when I open my IFC file and have the "InstanceCache" Null reference exception from EsentModel.cs, Visual is trying to find the .cs file in this following location "D:\a\1\s\Xbim.IO.Esent\Esent\EsentModel.cs" which is not a valid location at all, and I have no idea where it comes from and why this strange path exist in the first place. 

Do you think this Null reference exception could be related to this path "behavior" ?

**Comment by martin1cerny** [comment]
No, the path you see it the path from the build server. It is simply a stack record from the exception. 

I'm sure you will appreciate that we can't help you, if we can't reproduce the issue. There are hundreds to thousands working deployments of xbim Toolkit worldwide. So, this is likely an environmental issue.
